sync/class.jetpack-sync-listener.php

```diff
@@ -51,27 +52,41 @@ function get_sync_queue() {
 		return $this->sync_queue;
 	}
 
-	function set_queue_limit( $limit ) {
-		$this->sync_queue_limit = $limit;
+	function set_queue_size_limit( $limit ) {
+		$this->sync_queue_size_limit = $limit;
 	}
 
-	function get_queue_limit() {
-		return $this->sync_queue_limit;
+	function get_queue_size_limit() {
+		return $this->sync_queue_size_limit;
+	}
+
+	function set_queue_lag_limit( $age ) {
+		$this->sync_queue_lag_limit = $age;
+	}
+
+	function get_queue_lag_limit() {
+		return $this->sync_queue_lag_limit;
 	}
 
 	function force_recheck_queue_limit() {
-		delete_transient( self::QUEUE_SIZE_CHECK_TRANSIENT );
+		error_log("deleting transient ".self::QUEUE_STATE_CHECK_TRANSIENT);
+		delete_transient( self::QUEUE_STATE_CHECK_TRANSIENT );
 	}
 
-	function is_over_queue_limit() {
-		$queue_size = get_transient( self::QUEUE_SIZE_CHECK_TRANSIENT );
+	function can_add_to_queue() {
+		$queue_state = get_transient( self::QUEUE_STATE_CHECK_TRANSIENT );
 
-		if ( $queue_size === false ) {
+		if ( $queue_state === false ) {
 			$queue_size = $this->sync_queue->size();
-			set_transient( self::QUEUE_SIZE_CHECK_TRANSIENT, $queue_size, self::QUEUE_SIZE_CHECK_TIMEOUT );
+			$queue_age = $this->sync_queue->lag();
+			set_transient( self::QUEUE_STATE_CHECK_TRANSIENT, array( $queue_size, $queue_age ), self::QUEUE_STATE_CHECK_TIMEOUT );
```

Comment:
My thinking was this: If we're already prepared to wait up to 15 minutes without sending something before we check the queue size, then another 5 (maximum) is probably not going to cause additional breakage. 

Obviously this can be tuned during the beta period - and we probably should!